src/test/java/pl/piekoszek/gorskimatches/facebook/FacebookControllerTest.java

```diff
@@ -0,0 +1,205 @@
+                                "recipient":{
+                                    "id":"101650416067938"},
+                                "timestamp":1673444242999,
+                                "message":{
+                                "mid":"m_W6wvCjMgU-OuVylXuH5Iop5x1RXIqJ5aTf5qnNKQgJ-5o5iTO8njIWduCQ1jQ2INh_kJcw2sAIDunbDeO-0k4w",
+                                "text":"challenge"
+                                }}]}]}
+                             """)
+                .contentType(MediaType.APPLICATION_JSON)
+                .accept(MediaType.APPLICATION_JSON));
+        mockMvc.perform(post("/api/webhook/facebook/page/message").content("""
+                        {"entry":[{
+                            "id":"101650416067938",
+                            "time":1673444243864,
+                                "messaging":[{
+                                "sender":{
+                                    "id":"5533019560146899"},
+                                "recipient":{
+                                    "id":"101650416067938"},
+                                "timestamp":1673444242999,
+                                "message":{
+                                "mid":"m_W6wvCjMgU-OuVylXuH5Iop5x1RXIqJ5aTf5qnNKQgJ-5o5iTO8njIWduCQ1jQ2INh_kJcw2sAIDunbDeO-0k4w",
+                                "text":"7-3=4"
+                                }}]}]}
+                             """)
+                .contentType(MediaType.APPLICATION_JSON)
+                .accept(MediaType.APPLICATION_JSON));
+        Mockito.verify(facebookMessageService).sendReply("5533019560146899", "Great Job! For another quiz, type in: challenge");
+    }
+
+    @Test
+    public void shouldSendReplyWithTryAgainMessageWhenQuizIsCorrect() throws Exception {
+        Mockito.when(equationRandomizer.randomEquation()).thenReturn("7-5=4");
+        mockMvc.perform(post("/api/webhook/facebook/page/message").content("""
+                        {"entry":[{
+                            "id":"101650416067938",
+                            "time":1673444243864,
+                                "messaging":[{
+                                "sender":{
+                                    "id":"5533019560146899"},
+                                "recipient":{
+                                    "id":"101650416067938"},
+                                "timestamp":1673444242999,
+                                "message":{
+                                "mid":"m_W6wvCjMgU-OuVylXuH5Iop5x1RXIqJ5aTf5qnNKQgJ-5o5iTO8njIWduCQ1jQ2INh_kJcw2sAIDunbDeO-0k4w",
+                                "text":"challenge"
+                                }}]}]}
+                             """)
+                .contentType(MediaType.APPLICATION_JSON)
+                .accept(MediaType.APPLICATION_JSON));
+        mockMvc.perform(post("/api/webhook/facebook/page/message").content("""
+                        {"entry":[{
```

Comment:
If only difference between those JSONs is `"text"` field, you could create method with `String text` parameter which would create such JSON.